javav2/example_code/s3/src/main/java/com/example/s3/batch/CloudFormationHelper.java

```diff
@@ -0,0 +1,128 @@
+import software.amazon.awssdk.services.cloudformation.model.Capability;
+import software.amazon.awssdk.services.cloudformation.model.CloudFormationException;
+import software.amazon.awssdk.services.cloudformation.model.DescribeStacksRequest;
+import software.amazon.awssdk.services.cloudformation.model.DescribeStacksResponse;
+import software.amazon.awssdk.services.cloudformation.model.Stack;
+import software.amazon.awssdk.services.cloudformation.model.Output;
+import software.amazon.awssdk.services.cloudformation.waiters.CloudFormationAsyncWaiter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+public class CloudFormationHelper {
+
+    private static CloudFormationAsyncClient getCloudFormationClient() {
+        CloudFormationAsyncClient cfClient = CloudFormationAsyncClient.builder()
+            .region(Region.US_EAST_1)
+            .build();
+
+        return cfClient;
+    }
+
+    public static void deployCloudFormationStack(String stackName) {
+        String templateBody;
+        boolean doesExist = describeStack(stackName);
+        if (!doesExist) {
+            try {
+                Path filePath = Paths.get("template.yaml").toAbsolutePath();
+                templateBody = Files.readString(filePath);
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+
+            getCloudFormationClient().createStack(b -> b.stackName(stackName)
+                    .templateBody(templateBody)
+                    .capabilities(Capability.CAPABILITY_IAM))
+                .whenComplete((csr, t) -> {
+                    if (csr != null) {
+                        System.out.println("Stack creation requested, ARN is " + csr.stackId());
+                        try (CloudFormationAsyncWaiter waiter = getCloudFormationClient().waiter()) {
+                            waiter.waitUntilStackCreateComplete(request -> request.stackName(stackName))
+                                .whenComplete((dsr, th) -> {
+                                    if (th != null) {
+                                        System.out.println("Error waiting for stack creation: " + th.getMessage());
+                                    } else {
+                                        dsr.matched().response().orElseThrow(() -> new RuntimeException("Failed to deploy"));
+                                        System.out.println("Stack created successfully");
+                                    }
+                                }).join();
+                        }
+                    } else {
+                        System.out.format("Error creating stack: " + t.getMessage(), t);
+                        throw new RuntimeException(t.getCause().getMessage(), t);
```

Comment:
Handle this error if instance of `CloudFormationException`

Reply:
This is not a valid exception to throw back. Its protected. I looked into that